### ISSUE TITLE
Add Documentation for LabelEncoder

### ISSUE THREAD

**Issue by ashwinbhaskar** [picnicml/doddle-model#73]
It's difficult to understand what `LabelEncoder` does. I have tried  ooking at the method `inferLabelEncoder` in the `CsvLoader.scala`, to be specific the line 
```
encoder(name)(featureValue) = encoder(name).size
```
But it's not yet clear what `LabelEncoder` is being used for. 

Having  a comment documentation would really help here.

**Comment by inejc** [comment]
Good point, thanks.

**Comment by ashwinbhaskar** [comment]
@inejc is there any explanation that I can find somewhere? if you are kind of occupied, you can post an explanation here. I will try to comprehend it and add it into into the comments in the code. Does that work?

**Comment by inejc** [comment]
I am indeed busy. I appreciate the effort, thanks! Label encoder is described [here](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.LabelEncoder.html) and some examples can be found [here](https://scikit-learn.org/stable/modules/preprocessing_targets.html#label-encoding). Let me know if you need any more info.

**Comment by ashwinbhaskar** [comment]
@inejc Can you review the PR please?
